### PR TITLE
Display player name in all game modes when possible

### PR DESCRIPTION
**Pull request by AlexNisnevich**
#678, #599.

### CODE REVIEW

File: src/common/containers/App.js [hunk 1]
```diff
@@ -102,17 +103,20 @@ class App extends Component {
...
       return null;
     } else {
       return <NavMenu onRerender={this.props.onRerender} />;
     }
   }
 
   get content() {
-    const { inGame } = this.props;
-    const sidebarWidth = inGame ? 0 : (isFlagSet('sidebarCollapsed') ? SIDEBAR_COLLAPSED_WIDTH : SIDEBAR_WIDTH);
+    const sidebarWidth = this.inGame ? 0 : (isFlagSet('sidebarCollapsed') ? SIDEBAR_COLLAPSED_WIDTH : SIDEBAR_WIDTH);
```
Comment (john-patterson) [inline src/common/containers/App.js]: Should be broken into two lines:
```javascript
const sidebarWidthSetting = isFlagSet('sidebarCollapsed') ? SIDEBAR_COLLAPSED_WIDTH : SIDEBAR_WIDTH;
const sidebarWidth = this.inGame ? 0 : sidebarWidthSetting;
```

File: src/common/containers/GameAreaContainer.js [hunk 2]
```diff
@@ -216,33 +223,34 @@ export class GameAreaContainer extends Component {
     if (!started) {
       if (this.urlMatchesGameMode('tutorial')) {
         onStartTutorial();
-      } else if (this.urlMatchesGameMode('practice')) {
-        this.tryToStartPracticeGame(match.params.deck);
-      } else if (location.pathname.startsWith('/sandbox')) {
-        onStartSandbox();
+      } else if (firebaseLoaded) {
+        // Wait until we retrieve data from Firebase so we can get
```
Comment (john-patterson) [inline src/common/containers/GameAreaContainer.js]: The body of this if block is too large to be nested, should receive its own function.

File: src/common/containers/GameAreaContainer.js [hunk 3]
```diff
@@ -216,33 +223,34 @@ export class GameAreaContainer extends Component {
...
+        }
       } else {
-        history.push(Play.baseUrl);
+        // If we're still waiting on Firebase, render a message.
+        this.setState({ message: 'Connecting...' });
       }
     }
   };
 
   /* Try to start a practice game from the URL, pending Firebase loading. */
```
Comment (john-patterson) [inline src/common/containers/GameAreaContainer.js]: It seems like this comment is now lying about this function.

File: src/common/components/game/PlayerArea.js [hunk 4]
```diff
@@ -85,7 +85,7 @@ export default class PlayerArea extends Component {
         <PlayerName
           opponent={opponent}
           color={color}
-          playerName={(gameProps.player === color && !gameProps.isSandbox) ? 'You' : gameProps.usernames[color]} />
+          playerName={gameProps.usernames[color]} />
```
Comment (john-patterson) [inline src/common/components/game/PlayerArea.js]: Is the `isSandbox` case properly handled now?

Comment (AlexNisnevich) [reply src/common/components/game/PlayerArea.js]: Yeah, `usernames` should always be passed in now.

**Pull request merged**